src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/DfaMatchingState.cs

```diff
@@ -55,11 +56,15 @@ internal int WatchDog
         internal bool StartsWithLineAnchor => Node._info.StartsWithLineAnchor;
 
         /// <summary>
-        /// Compute the target state for the given input minterm.
-        /// If <paramref name="minterm"/> is False this means that this is \n and it is the last character of the input.
+        /// Translates a minterm predicate to a character kind, which is a general categorization of characters used
+        /// for cheaply deciding the nullability of anchors.
         /// </summary>
-        /// <param name="minterm">minterm corresponding to some input character or False corresponding to last \n</param>
-        internal DfaMatchingState<T> Next(T minterm)
+        /// <remarks>
+        /// A False predicate is handled as a special case to indicate the very last \n.
+        /// </remarks>
+        /// <param name="minterm">the minterm to translate</param>
+        /// <returns>the character kind of the minterm</returns>
+        private uint GetNextCharKind(ref T minterm)
```

Comment:
Is there a reason why the return type is `uint` as opposed to `CharKind`?

Reply:
One reason is simplicity in treatinig a kind the same way as a pair of kinds (context), 
so` uint Context(uint prevKind, uint nextKind)` takes two kinds but pairs them (by shifting one 3 bits to the left) together into one `uint` that is essentially then 'a pair' of kinds. The operations `Next` and `Prev` hide these details.

`CharKindCount` is the number of values relevant for a single kind (5 to be precise, values 0 through 4)

This encoding helped keep the cost at minimum because it is used everywhere in hot-paths of code, to carry previous and next character kinds.